ffi/ffi_wrapper.go

```diff
@@ -312,6 +313,66 @@ func CreateTwinDidWithControlDelegation(
 	return C.CString(twinIdentity.Did()), nil
 }
 
+//export GetPrivateExponentHexFromPrivateKeyBase64
+func GetPrivateExponentHexFromPrivateKeyBase64(cPrivateKeyBase64 *C.char) (*C.char, *C.char) {
+	// convert `*C.char` to `string`
+	privateKeyBase64 := C.GoString(cPrivateKeyBase64)
+
+	privateExponentHex, err := advancedapi.GetPrivateExponentHexFromPrivateKeyBase64(privateKeyBase64)
+	if err != nil {
+		return nil, C.CString(fmt.Sprintf("FFI lib error: failed to get private exponent hex from private key base64: %+v", err))
+	}
+
+	return C.CString(privateExponentHex), nil
+}
+
+//export TwinDelegatesControlToAgentByPrivateExponentHex
+func TwinDelegatesControlToAgentByPrivateExponentHex(
+	cResolverAddress *C.char,
+	cAgentDid *C.char,
+	cAgentKeyName *C.char,
+	cAgentName *C.char,
+	cAgentSeed *C.char,
+	cPrivateExponentHex *C.char,
+	cDelegationName *C.char) *C.char {
+	// convert `*C.char` to `string`
+	resolverAddress := C.GoString(cResolverAddress)
+	agentDid := C.GoString(cAgentDid)
+	agentKeyName := C.GoString(cAgentKeyName)
+	agentName := C.GoString(cAgentName)
+	agentSeed := C.GoString(cAgentSeed)
+	privateExponentHex := C.GoString(cPrivateExponentHex)
+	delegationName := C.GoString(cDelegationName)
+
+	// Resolver
+	addr, err := url.Parse(resolverAddress)
+	if err != nil {
+		return C.CString(fmt.Sprintf("FFI lib error: parsing resolver address failed: %+v", err))
+	}
+	resolver := register.NewDefaultRestResolverClient(addr)
+
+	// Agent Identity
+	agentIdentity, _, err := getIdentity(api.GetAgentIdentity, agentDid, agentKeyName, agentName, agentSeed)
```

Comment:
The ffi wrapper should not call/add any logic that is not in the API. 
The wrappers should transparently call the the API helpers

Reply:
The ffi wrapper should transparently call the API. 
```
// API code
func TheAPIFunction(arg1, arg2 string) { 
...
}

// FFI code
func TheFFICode(arg1 *C.char, arg2 *C.char) {
  // no logic
  TheAPIFunction(C.GoString(arg1), C.GoString(arg2))
}
```

https://github.com/Iotic-Labs/iotics-identity-go/blob/72bf190cda49e677e5c792054b718b104870bb3c/ffi/ffi_wrapper.go#L366 contains logic

Reply:
But that's not part of this PR. Maybe it would sit best as a tech debt story?

Reply:
parameter validation is ok in wrappers IMO
I think SB is referring to combining multiple calls to the advanced API in a single wrapper call.
And I agree - if it's in the wrapper it's because is useful so having this function in the high level api golang seems more appropriate. And it'll make the code simpler.

Reply:
If the takeaway here is that both the identity API and the wrapper can use a rethink, I'm fully behind that.

I've also tried to raise the alarm in the comments here that the function signatures of `DelegateControlByPrivateExponentHex` and `TakeOwnershipOfTwinByPrivateExponentHex` might also need a rethink. I am still waiting for a response on that.

However, the code reviewing of untouched code and the expectation that I can magically fix all the things that in reality will require a lot of heated debates for which there is no scheduled time currently, makes me question the real intention behind all this.